### ISSUE TITLE
JOmniFactory should allow for specifying that PodioOutput is a subset collection

### ISSUE THREAD

**Issue by wdconinc** [eic/EICrecon#1169]
**Is your feature request related to a problem? Please describe.**
Conversion of a few factories to JOmniFactory is held up by the inability to specify that the output collection is a subset collection, i.e. #1168 for details about which ones are affected.

**Describe the solution you'd like**
It could be as simple as:
```
    PodioOutput<edm4eic::Cluster> m_cluster_output {this, "Reconstructed clusters", bool owns_data = false};
```
similar to how this is done with `DeclarePodioOutput`.

**Describe alternatives you've considered**
This can also be argued to be an algorithm responsibility, so it would have to start with:
```
  void CalorimeterClusterRecoCoG::process(
      const CalorimeterClusterRecoCoG::Input& input,
      const CalorimeterClusterRecoCoG::Output& output) const {

    const auto [proto, mchits] = input;
    auto [clusters, associations] = output;

    clusters.setSubsetCollection();
```
but that means that JANA doesn't know about whether to clean up the memory...

**Comment by wdconinc** [comment]
@nathanwbrei Did you have any plans as to where to take the `PodioOutput` interface for subset collections that don't own their data?

@sly2j Where is algorithms/juggler/gaudi specifying that something is a subset collection?

**Comment by nathanwbrei** [comment]
Yes, I have some work in the pipeline on this front. Am I correct in thinking that the only place we need this for now is in `TrackerHitCollector_factory`?

**Comment by wdconinc** [comment]
Also `ChargedParticleSelector_factory`. See first list at #1168.
